packages/docusaurus-theme-classic/src/theme/DocSidebarItem/Link/index.tsx

```diff
@@ -23,10 +23,15 @@ export default function DocSidebarItemLink({
   level,
   index,
   ...props
-}: Props): JSX.Element {
-  const {href, label, className, autoAddBaseUrl} = item;
+}: Props): JSX.Element | null {
+  const {href, label, className, autoAddBaseUrl, unlisted} = item;
   const isActive = isActiveSidebarItem(item, activePath);
   const isInternalLink = isInternalUrl(href);
+
+  if (unlisted && !isActive) {
+    return null;
+  }
```

Comment:
same, call `isInactiveUnlistedSidebarItem(item, activePath)` upper in the tree, make sure `DocSidebarItemLink` does not render for inactive unlisted doc